### PR TITLE
[AC-5519] Add testing and coverage

### PR DESCRIPTION
**Pull request by jonkiparsky**
So - this works, with a big "but" that we need to deal with

The module name "app-allocator-scripts" is not a valid python module name, so it breaks imports. If you rename the directory as "app_allocator_scripts", or if you do a new clone of the repo into a directory with that name, this works, but if you've done the standard git clone into the default-named directory then this will fail.

A few options to choose from:
1) rename the repo so a default checkout will Just Work
2) require users to clone into app_allocator_scripts
3) introduce a new level of hierarchy so we'd have app-allocator-scripts/src/classes

@mc-nathan I think the last is the most standard, but I'm open to any thoughts you have. I don't have a strong preference in this case. 

That aside, this is pretty straightforward. Changes introduced in this PR:
1) added make targets for `test` and `coverage`
2) added a single test for judge.py

Note: since we are not working in a container or a virtualenv, it is assumed that users have pytest and pytest-cov installed (use pip to install)

Also, note that pytest-cov is only reporting coverage on files imported in tests, and I have not yet worked out how to skip files not touched. 
Hopefully AC-5520 will get 100% coverage on all of the files in /classes and it'll all be just nice and we won't have to worry too much about this.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,3 +17,11 @@ help:
 code-check:
 	@pycodestyle *.py */*.py
 	@flake8 *.py */*.py
+
+test:
+	@pytest
+
+
```
Comment (mc-nathan) [inline Makefile]: Minor consistency point.  I would tend to either use a single space between all the targets, or I would group test and coverage together and put the double space above test.

File: app_allocator/Makefile [hunk 2]
```diff
@@ -17,3 +17,11 @@ help:
 code-check:
 	@pycodestyle *.py */*.py
 	@flake8 *.py */*.py
+
+test:
```
Comment (mc-nathan) [inline app_allocator/Makefile]: Add to `make help`

File: app_allocator/Makefile [hunk 3]
```diff
@@ -17,3 +17,11 @@ help:
...
+	@pytest
+
+
+coverage:
```
Comment (mc-nathan) [inline app_allocator/Makefile]: Add to `make help`

File: app_allocator/classes/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (mc-nathan) [inline app_allocator/classes/__init__.py]: flake8 error:
```
$ make code-check
classes/__init__.py:1:1: W391 blank line at end of file
make: *** [code-check] Error 1
```

**Review by mc-nathan — CHANGES_REQUESTED** [review]
Need help for new make targets and minor flake8 issue.

**Comment by jonkiparsky** [comment]
> One more thing, app_allocator/.pytest_cache/ should get added to .gitignore.

**Comment by mc-nathan** [comment]
Functional issue, something is not right with the imports:
```
(venv) MC-HQ-NWILSON:app-allocator-scripts nathan$ python3 app_allocator/allocate.py example.csv ordered_queue > oq.out
Traceback (most recent call last):
  File "app_allocator/allocate.py", line 10, in <module>
    from classes.allocator import Allocator
  File "/Users/nathan/src/app-allocator-scripts/app_allocator/classes/allocator.py", line 14, in <module>
    from classes.judge import Judge
  File "/Users/nathan/src/app-allocator-scripts/app_allocator/classes/judge.py", line 2, in <module>
    from app_allocator.classes.entity import Entity
ModuleNotFoundError: No module named 'app_allocator'
(venv) MC-HQ-NWILSON:app-allocator-scripts nathan$ python3 app_allocator/allocate.py example.csv > tmp1.out
Traceback (most recent call last):
  File "app_allocator/allocate.py", line 10, in <module>
    from classes.allocator import Allocator
  File "/Users/nathan/src/app-allocator-scripts/app_allocator/classes/allocator.py", line 14, in <module>
    from classes.judge import Judge
  File "/Users/nathan/src/app-allocator-scripts/app_allocator/classes/judge.py", line 2, in <module>
    from app_allocator.classes.entity import Entity
```

File: Makefile (hunk 1 above)
Comment (jonkiparsky) [reply Makefile]: ✅

File: app_allocator/Makefile (hunk 2 above)
Comment (jonkiparsky) [reply app_allocator/Makefile]: ✅

File: app_allocator/Makefile (hunk 3 above)
Comment (jonkiparsky) [reply app_allocator/Makefile]: ✅

File: app_allocator/classes/__init__.py (hunk 4 above)
Comment (jonkiparsky) [reply app_allocator/classes/__init__.py]: This line doesn't seem to exist except in code-check's head. It doesn't show in emacs and I can't delete it through the github editor

**Comment by jonkiparsky** [comment]
@mc-nathan the current structure allows all of the scripts to run, but it looks like we'll have a hard time getting the scripts themselves under test. This is not terribly alarming to me, since they're not really the point of the whole business and I'm sure we can work out a way to make the work if it becomes important, but it seems worth mentioning. Anyway, there it is.

File: app_allocator/classes/__init__.py (hunk 4 above)
Comment (mc-nathan) [reply app_allocator/classes/__init__.py]: I was able to get rid of it and pushed the change.  code-check is happy in my local.

**Review by mc-nathan — APPROVED** [review]

**Pull request merged**